pkg/config/config.go

```diff
@@ -2,88 +2,105 @@ package config
-	viper.BindEnv(constants.ConfigHost, constants.EnvOctopusHost)
-	viper.BindEnv(constants.ConfigSpace, constants.EnvOctopusSpace)
+func bindEnvironment(v *viper.Viper) error {
+	if err := v.BindEnv(constants.ConfigApiKey, constants.EnvOctopusApiKey); err != nil {
+		return err
+	}
+	if err := v.BindEnv(constants.ConfigHost, constants.EnvOctopusHost); err != nil {
+		return err
+	}
+	if err := v.BindEnv(constants.ConfigSpace, constants.EnvOctopusSpace); err != nil {
+		return err
+	}
 	// Envs will take precedence in the specified order
-	viper.BindEnv(constants.ConfigEditor, constants.EnvVisual, constants.EnvEditor)
-	viper.BindEnv(constants.ConfigNoPrompt, constants.EnvCI)
+	if err := v.BindEnv(constants.ConfigEditor, constants.EnvVisual, constants.EnvEditor); err != nil {
+		return err
+	}
+	if err := v.BindEnv(constants.ConfigNoPrompt, constants.EnvCI); err != nil {
+		return err
+	}
+	return nil
 }
 
-// CreateNewConfig will create a config file and read it if it does not
-// yet exist
-func CreateNewConfig() error {
-	if err := writeNewConfig(); err != nil {
-		if _, ok := err.(viper.ConfigFileAlreadyExistsError); ok {
-			return nil
-		}
-		return fmt.Errorf("error writing config file: %w", err)
+func Setup(v *viper.Viper) error {
+	setDefaults(v)
+	if err := bindEnvironment(v); err != nil {
+		return err
 	}
-	if err := viper.ReadInConfig(); err != nil {
-		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
-			return fmt.Errorf("error could not find config file after create: %w", err)
-		} else {
-			// Config file was found but something is wrong
-			return fmt.Errorf("error reading config file after create: %w", err)
+
+	configPath, err := getConfigPath()
+	if err == nil {
+		SetupConfigFile(v, configPath)
+		if err := v.ReadInConfig(); err != nil {
+			if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+				// Do nothing, config file will be created on `config set` cmd
+				// This is to avoid issues with CI tools that may not have access
+				// to the file system
+			} else {
+				// Config file was found but something is wrong
+				// we can recover and run with no config
+				fmt.Println("Error reading config file: %w", err)
+			}
 		}
 	}
+	// if we can't get the configPath, then everything will just be defaulted
 	return nil
 }
 
+// EnsureConfigPath works out the config path, then creates the directory to make sure that it exists
+func EnsureConfigPath() (string, error) {
+	configPath, err := getConfigPath()
+	if err != nil {
+		return "", err
+	}
+	err = os.MkdirAll(configPath, os.ModePerm)
+	if err != nil {
+		return "", err
+	}
+	return configPath, nil
+}
+
+// getConfigPath works out the directory where the config file should be saved and returns it.
+// does not modify the global viper
 func getConfigPath() (string, error) {
 	if runtime.GOOS == "windows" {
 		if appdataPath := os.Getenv(appData); appdataPath != "" {
 			configPath := filepath.Join(appdataPath, "octopus")
-			viper.AddConfigPath(configPath)
```

Comment:
Burying the side-effect of setting the config path on the global viper instance made this pretty hard to figure out :-(